src/components/Inventory/withdraw/flow/WithdrawalBatchesCellRenderer.tsx

```diff
@@ -0,0 +1,150 @@
+import React from 'react';
+import { Link } from 'react-router-dom';
+import { makeStyles } from '@mui/styles';
+import { Box, Typography, useTheme } from '@mui/material';
+import { APP_PATHS } from 'src/constants';
+import { CellRenderer, RendererProps, TableRowType } from '@terraware/web-components';
+import TextField from '@terraware/web-components/components/Textfield/Textfield';
+
+const useStyles = makeStyles(() => ({
+  text: {
+    fontSize: '14px',
+    '& > p': {
+      fontSize: '14px',
+    },
+  },
+}));
+
+export default function WithdrawalBatchesCellRenderer(props: RendererProps<TableRowType>): JSX.Element {
+  const classes = useStyles();
+  const theme = useTheme();
+  const { column, row, value, index, onRowClick } = props;
+
+  const createLinkToBatchDetail = (iValue: React.ReactNode | unknown[]) => {
+    return (
+      <Link
+        to={`${APP_PATHS.INVENTORY_ITEM.replace(':speciesId', row.speciesId.toString())}?batch=${iValue}`}
+        style={{ color: theme.palette.TwClrTxtBrand }}
+        className={classes.text}
+      >
+        {iValue}
+      </Link>
+    );
+  };
+
+  const createNotReadyInput = (iValue: React.ReactNode | unknown[]) => {
+    if (onRowClick) {
+      return (
+        <Box display='flex' alignItems='center'>
+          <TextField
+            id='notReadyQuantityWithdrawn'
+            type='text'
+            onChange={(id, value) => onRowClick(value as string)}
+            value={row.notReadyQuantityWithdrawn}
+            label={''}
+          />
+          <Typography paddingLeft={1}>/ {row.notReadyQuantity} </Typography>
+        </Box>
+      );
+    }
+  };
+
+  const createReadyInput = (iValue: React.ReactNode | unknown[]) => {
+    if (onRowClick) {
+      return (
+        <Box display='flex' alignItems='center'>
+          <TextField
+            id='readyQuantityWithdrawn'
+            type='text'
+            onChange={(id, value) => onRowClick(value as string)}
+            value={row.readyQuantityWithdrawn}
+            label={''}
+          />
+          <Typography paddingLeft={1}>/ {row.readyQuantity} </Typography>
+        </Box>
+      );
+    }
+  };
+
+  const createReadyOutplantInput = (iValue: React.ReactNode | unknown[]) => {
+    if (onRowClick) {
+      return (
+        <TextField
+          id='readyQuantityWithdrawn'
+          type='text'
+          onChange={(id, value) => onRowClick(value as string)}
+          value={row.readyQuantityWithdrawn}
+          label={''}
+        />
```

Comment:
we could extract this as a function and use it in all the ready/not-ready input places..
```
  const createQuantityInput = (id, valueProperty) => (
     <TextField
         id={id}
         type='text'
         onChange={(_, value) => onRowClick(value as string)
         value={row[valueProperty]
          label={''}
      />
  );
```